ci-operator/config/codeready-toolchain/toolchain-e2e/codeready-toolchain-toolchain-e2e-master.yaml

```diff
@@ -18,7 +18,7 @@ resources:
       memory: 250Mi
 tests:
 - as: e2e
-  skip_if_only_changed: ^docs/|^\.github/|^\.tekton/|\.md$|\.adoc$|\.properties$|^(?:.*/)?(?:\.gitignore|.codecov.yml|.yamllint|OWNERS|PROJECT|LICENSE)$
+  skip_if_only_changed: ^doc/|\\.(md|adoc)$|^(?:.*/)?(?:\.gitignore|.codecov.yml|.yamllint|sonar-project.properties|OWNERS|PROJECT|LICENSE)$
```

Comment:
why not
```suggestion
  skip_if_only_changed: ^doc/|\.md$|\.adoc$|^(?:.*/)?(?:\.gitignore|.codecov.yml|.yamllint|sonar-project.properties|OWNERS|PROJECT|LICENSE)$
```

Reply:
and if you want to keep the original one - are you sure about the two back-slashes?

Reply:
could you please check the regex one more time - some of the parts look weird to me:
* you don't escape the forward slashes
* you don't escape the dots in .codecov.yml, .yamllint, and sonar-project.properties
* codecov can be also in the format .codecov.yaml (which is what we use in our projects)
* could you please explain why you need to use two non capturing groups?
* why the first part `^doc/|\.md$|\.adoc$` is not included in the group of the second part? 
* the beginning of the line char `^` seems to be redundant in front of the first non capturing group expression
* the `\.md$|\.adoc$|` matches only the suffixes, not the entire file names - is this sufficient? if so, then why we need to non capturing groups in the second part at all? 

TBH, I'm not good at regexes, so please correct me if I'm missing some important detail

Reply:
both are fine  ref: https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes

Reply:
> * you don't escape the dots in .codecov.yml, .yamllint, and sonar-project.properties
> * codecov can be also in the format .codecov.yaml (which is what we use in our projects)

good one i will modify it to support yml | yaml

Reply:
> both are fine ref: https://docs.prow.k8s.io/docs/jobs/#triggering-jobs-based-on-changes

I doubt that the version with two backslashes would be fine. Two backslashes result into one backslash, so you would expect the files to end with: `\.md`

Just to be sure, I tried it in go playground:

https://go.dev/play/p/zuVWv_jd_UI